api/src/processing/services/bilibili.js

```diff
@@ -17,8 +17,12 @@ function extractBestQuality(dashData) {
     return [ bestVideo, bestAudio ];
 }
 
-async function com_download(id) {
-    let html = await fetch(`https://bilibili.com/video/${id}`, {
+async function com_download(id , episode = 1) {
+    let fetchUrl = `https://bilibili.com/video/${id}`;
+    if (episode > 1) {
+        fetchUrl += `?p=${episode}`
+    }
+    let html = await fetch(fetchUrl, {
```

Comment:
```suggestion
    const html = await fetch(`https://bilibili.com/video/${id}?p=${episode}`, {
```

1. comparing a string (`episode`) to a number won't work the way you expect it to.
2. it's fine to always use the `p` query parameter in the fetch url.